docker-maven-plugin/src/main/java/org/openmrs/maven/plugins/AbstractDockerMojo.java

```diff
@@ -54,26 +56,32 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     public abstract void executeTask() throws MojoExecutionException, MojoFailureException;
 
     protected void resolveDocker() {
-        if(dockerHost == null){
-            boolean isLinux = SystemUtils.IS_OS_LINUX;
-            if(isLinux){
-                dockerHost = prompt("Please specify your Docker host URL (either 'tcp://' or 'unix://')", DEFAULT_HOST_LINUX);
-            } else {
-                dockerHost = prompt("Please specify you Docker Machine host URL (format is: 'tcp://{docker-machine url}')","");
-            }
-        }
 
-        DockerClientConfig config = DockerClientConfig.createDefaultConfigBuilder()
+        DockerClientConfig config = DefaultDockerClientConfig.createDefaultConfigBuilder()
                 .withDockerHost(dockerHost)
-                .withApiVersion("1.18")
+                .withApiVersion(API_VERSION)
                 .build();
 
         docker = DockerClientBuilder.getInstance(config).build();
 
         try {
             docker.infoCmd().exec();
         } catch (Exception e) {
-            String message = "Could not run `docker info`. If you are using 'Docker Toolbox', please make sure you run the SDK command from the 'Docker Toolbox' terminal.";
+            String message;
+            if (dockerHost.equals(DEFAULT_HOST_LINUX)) {
+                message = "Couldn't connect to docker under default docker host address for linux";
+                boolean isUserWantChangeUrl = promptYesNo("Do You want to change docker host address?");
+                if (isUserWantChangeUrl) {
+                    dockerHost = prompt("Type updated host url (You can save this property later by calling openmrs-sdk:setup with defined -DdockerHost)");
+                    resolveDocker();
+                }
+                else {
+                    throw new RuntimeException(message, e);
+                }
+            }
+            else {
+                message = "Could not run `docker info`. If you are using 'Docker Toolbox', please make sure you run the SDK command from the 'Docker Toolbox' terminal.";
+            }
```

Comment:
Please change this behavior as follows:

1. IF not Linux: 
Could not connect to Docker at dockerHostUrl. If you are using 'Docker Toolbox', please make sure you run the SDK command from the 'Docker Toolbox' terminal. If the docker host URL is not correct, please specify correct URL by passing the -DdockerHost="tcp://..." parameter.

2. IF Linux:
Could not connect to Docker at dockerHostUrl. If the docker host URL is not correct, please specify new URL by passing the -DdockerHost="tcp://..." parameter.

In both cases we should stop further execution.